ui/src/components/Collection/CollectionDetail/TestMediaItem/index.tsx

```diff
@@ -168,6 +170,14 @@ const TestMediaItem = (props: ITestMediaItem) => {
     return
   }
 
+  const copyToClipboard = () => {
+    const value = (editorRef.current as any)?.getValue?.()
+    if (value?.trim()) {
+      navigator.clipboard.writeText(value)
+      toast.success('Copied to clipboard')
+    }
+  }
+
```

Comment:
So copying to clipboard is kind of a nightmare in certain situations, like in Firefox it is only available in "[Secure Contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts)", meaning it doesn't work for URLs like `http://maintainer.home.arpa`. This also appears to be the same for Chrome.

You could check if `navigator.clipboard` is not undefined at the top of the component, and then toggle the button display based on that.

`const isClipboardAvailable = navigator.clipboard !== undefined`

There are other hacks to workaround around this issue if you wanted to go down that route, such as: https://stackoverflow.com/a/65996386

`navigator.clipboard.writeText` can also fail so it's worth wrapping in a try catch and having an error toast to handle that case:

```suggestion
  const copyToClipboard = async () => {
    const value = (editorRef.current as any)?.getValue?.()
    if (value?.trim()) {
	try {
            await navigator.clipboard.writeText(value)
            toast.success('Copied to clipboard')
        } catch (error) {
            toast.error('Failed to copy to clipboard')
        }
    }
  }

```